workerpool/pool.go

```diff
@@ -207,11 +212,18 @@ func (w *worker) run() {
 			}
 			return
 		default:
-			data, err = w.connection.Source.Fetch(w.id, data.LastSequence)
+			checkpoint, err = w.checkpointKV.RetrieveCheckpoint(w.checkpointID)
+			if err != nil && err != kv.ErrKeyNotFound {
```

Comment:
this ErrKeyNotFound should be ignored inside `RetrieveCheckpoint`. It's an internal thing.

Reply:
it is ignored, if there's some other kind of failure we'd want to know about it. If it's just `ErrKeyNotFound` we skip

Reply:
The thing is that workerpool package has to be aware of `ErrKeyNotFound`. Which also means that it has to be away of some kind of Key (Err**Key**NotFound). It creates unneeded dependency.

We can simplify that code

```
if err != nil {
...
}
data, err = w.connection.Source.Fetch(w.id, checkpoint)
```
And in case of `ErrKeyNotFound` just return `"", nil` instead of `"", err"` from `RetrieveCheckpoint`.

It's not a big deal, just a small thing that can slightly improve the interface of `RetrieveCheckpoint`.